tools/run_tests/run_performance_tests.py

```diff
@@ -450,15 +451,21 @@ def finish_qps_workers(jobs):
 if not scenarios:
   raise Exception('No scenarios to run')
 
+num_failures = 0
 for scenario in scenarios:
   if args.dry_run:
     print(scenario.name)
   else:
     try:
       for worker in scenario.workers:
         worker.start()
-      jobset.run([scenario.jobspec,
-                  create_quit_jobspec(scenario.workers, remote_host=args.remote_driver_host)],
-                 newline_on_success=True, maxjobs=1)
+      jobset_failures, _ = jobset.run([scenario.jobspec,
+                                 create_quit_jobspec(scenario.workers, remote_host=args.remote_driver_host)],
+                                 newline_on_success=True, maxjobs=1)
+      num_failures += jobset_failures
     finally:
-      finish_qps_workers(scenario.workers)
+      # Consider jobs that need to be killed as failures
+      num_failures += finish_qps_workers(scenario.workers)
+
+if num_failures > 0:
```

Comment:
nit: maybe we can count   scenario_failures and number of killed workers separately and then report both if there's a failure.
`X scenarios failed and Y workers needed to be killed`